cmd/cluster-agent/app/app.go

```diff
@@ -195,7 +195,7 @@ func start(cmd *cobra.Command, args []string) error {
 	log.Infof("Got APIClient connection")
 
 	// Get hostname as aggregator requires hostname
-	hostname, err := util.GetHostname()
+	hostname, err := util.GetHostname(context.TODO())
```

Comment:
what is you plan about `context.TODO()`.

Reply:
So far, I’ve introduced a “real” `context` (i.e.: not a `context.TODO()`) in the components that are monitored by a health probe.

There are some cases where we cannot introduce a “real” `context` (i.e.: one with a finite timeout or deadline) without breaking the compatibility. That’s the case for checks for ex. The duration of a check execution is currently not bounded by a time-out.
So, if a check need to call the kubelet, the docker daemon or whatever, each individual call will be protected by an individual time-out, but there’s no global deadline for the whole check execution.
So, for checks, we might want to initialise the context with `context.Background()`.

There are also the case of user CLI commands like `agent status`, `agent flare`, etc.
In that situation, I may have left `context.TODO()`.
If we think that an end-user will, anyhow, not wait forever for an interactive command to answer if he hasn’t any evidence of progress, we may consider setting a “user interaction” time-out.
But that could be in another PR.
This one focused on components monitored by a health probe.